src/prefab_manager.zig

```diff
@@ -0,0 +1,197 @@
+const std = @import("std");
+const flecs = @import("flecs");
+const zm = @import("zmath");
+const zmesh = @import("zmesh");
+const zwin32 = @import("zwin32");
+
+const fd = @import("flecs_data.zig");
+const gfx = @import("gfx_d3d12.zig");
+const mesh_loader = @import("renderer/mesh_loader.zig");
+const util = @import("util.zig");
+const rt = @import("renderer/renderer_types.zig");
+const IdLocal = @import("variant.zig").IdLocal;
+
+const assert = std.debug.assert;
+const d3d12 = zwin32.d3d12;
+const zcgltf = zmesh.io.zcgltf;
+
+const PrefabHashMap = std.AutoHashMap(IdLocal, flecs.Entity);
+
+pub const PrefabManager = struct {
+    prefab_hash_map: PrefabHashMap,
+    is_a: flecs.Entity,
+
+    pub fn init(world: *flecs.World, allocator: std.mem.Allocator) PrefabManager {
+        return PrefabManager{
+            .prefab_hash_map = PrefabHashMap.init(allocator),
+            .is_a = flecs.Entity.init(world.world, flecs.c.Constants.EcsIsA),
+        };
+    }
+
+    pub fn deinit(self: *@This()) void {
+        self.prefab_hash_map.deinit();
+    }
+
+    pub fn loadPrefabFromGLTF(
+        self: *@This(),
+        path: [:0]const u8,
+        world: *flecs.World,
+        gfxstate: *gfx.D3D12State,
+        allocator: std.mem.Allocator
+    ) !flecs.Entity {
+        const path_id = IdLocal.init(path);
+        var existing_prefab = self.prefab_hash_map.get(path_id);
+        if (existing_prefab) |prefab| {
+            return prefab;
+        }
+
+        const data = try zmesh.io.parseAndLoadFile(path);
+        defer zmesh.io.freeData(data);
+
+        assert(data.scenes_count == 1);
+        const scene = &data.scenes.?[0];
+        assert(scene.nodes_count == 1);
+
+        var arena_state = std.heap.ArenaAllocator.init(allocator);
+        defer arena_state.deinit();
+        const arena = arena_state.allocator();
+
+        var prefab = world.newPrefab(path);
+        prefab.setOverride(fd.Position.init(0, 0, 0));
```

Comment:
In the future prefabs won't necessarily have positions etc. One could imagine a prefab for a quest or a faction for example. But it's ok for now :)